### ISSUE TITLE
How to export all list events?

### ISSUE THREAD

**Issue by thansau239** [MISP/MISP#819]
Dear all,
please help me, How to export all list events?
Thank you

**Comment by cvandeplas** [comment]
I am not sure to understand your question correctly, could you clarify what you are trying to do?

Exporting the event-list can be done in various ways:
- using an external script with PyMISP or directly using the API
- using the webinterface by manually printing the page ( there's a print specific stylesheet)
- using the webinterface by adding .json or .xml to the url, you should get the page you have in the chosen format.

**Comment by iglocska** [comment]
Depends on the format, but in all cases you will generally have two options:

1. use the cached exports (navigate to event actions -> exports)
2. use the API (navigate to event actions -> automation for a description)

From both of those you can choose the format in which you'd like to download the events in.

**Comment by thansau239** [comment]
For example: in this attach file: i have some event-list. Now, i want to export all event-list to csv file with all field as event-list
![2016-01-07_153058](https://cloud.githubusercontent.com/assets/13726639/12165870/05336b0c-b554-11e5-820a-dcfc109eb400.png)

**Comment by iglocska** [comment]
Just go to Export (Event actions -> Export), in the CSV row, click
generate. Once it is finished click download.

Alternatively, use the API https://yourmisp/events/csv/download

**Comment by thansau239** [comment]
@iglocska After downloaded, nothing in csv file
![2016-01-07_160557](https://cloud.githubusercontent.com/assets/13726639/12166466/c0a149fa-b558-11e5-85a5-415ca55a0138.png)

**Comment by iglocska** [comment]
In the events that you have, do you have any attributes marked as "to_ids"?
Exports in general (except for XML, JSON and STIX) ignore any attributes
that are not marked to_ids. You can override this via the API though.

**Comment by thansau239** [comment]
@iglocska  Thank you, so how to include some field( tags, Email, Date,Threat Level , Analysis, Info...) in extract file?

**Comment by iglocska** [comment]
For that you have to use the API (there is an add context flag), but that is a good point, maybe having the context in the cached full export would make sense. I'll try to add this soon.

**Comment by thansau239** [comment]
I tried export but no context in extract file. Please add this soon for summary info
Thank you

**Comment by advancedeng** [comment]
You are right Andras, IncludeContext in the cached full export does makes sense :)

**Comment by naisanza** [comment]
@iglocska using the API https://misp.gitbooks.io/misp-book/content/automation/#xml-export

Is it a `GET` or `POST` request?

**Comment by iglocska** [comment]
GET if you use url parameters, POST if you post the filters. However, the xml export is crap, use restSearch instead. We generally recommend JSON output over XML, but if you really want XML you can get XML output from restSearch.

**Comment by adulau** [comment]
CSV export of index has been added in the current version.  https://github.com/MISP/MISP/commit/1187fb2a27233bc15559971610931c772b8c6763 Feel free to test it.

**Comment by adulau** [comment]
CSV export of index has been added in the current version.  https://github.com/MISP/MISP/commit/1187fb2a27233bc15559971610931c772b8c6763 Feel free to test it.
